qcodes/instrument_drivers/Keysight/Keysight_81180A.py

```diff
@@ -53,360 +76,399 @@ def __init__(self, parent, name, id, **kwargs):
+            "trigger_input",
+            set_cmd="TRIGGER:{}",
+            vals=vals.Enum("ECL", "TTL"),
+            docstring="Set trigger input. Possible inputs are:\n"
+            "\tECL: negative signal at fixed -1.3V.\n"
+            "\tTTL: variable trigger_level",
+        )
+
+        self.add_parameter(
+            "trigger_source",
+            get_cmd="TRIGGER:SOURCE:ADVANCE?",
+            set_cmd="TRIGGER:SOURCE:ADVANCE {}",
+            vals=EnumCaseInsensitive("EXTernal", "BUS", "TIMer", "EVENt"),
+            docstring="Possible trigger sources are:\n"
+            "\texternal: Use TRIG IN port exclusively.\n"
+            "\tbus: Use remote commands exclusively.\n"
+            "\ttimer: Use internal trigger generator exclusively.\n"
+            "\tevent: Use EVENT IN port exclusively.",
+        )
+
+        self.add_parameter(
+            "trigger_mode",
+            get_cmd="TRIGGER:MODE?",
+            set_cmd="TRIGGER:MODE {}",
+            vals=EnumCaseInsensitive("NORMal", "OVERride"),
+            docstring="Possible trigger modes are:\n"
+            "\tnormal:  the first trigger activates the output and "
+            "consecutive triggers are ignored for the duration of "
+            "the output waveform.\n"
+            "\toverride:  the first trigger activates  the output "
+            "and consecutive triggers  restart the output waveform, "
+            "regardless if the current waveform has been completed "
+            "or not.",
+        )
+
+        self.add_parameter(
+            "trigger_timer_mode",
+            get_cmd="TRIGGER:TIMER:MODE?",
+            set_cmd="TRIGGER:TIMER:MODE {}",
+            vals=EnumCaseInsensitive("TIME", "DELay"),
+            docstring="Possible modes of internal trigger are:\n"
+            "\ttime: Perform trigger at fixed time interval.\n"
+            "\tdelay: Perform trigger at fixed time intervals after "
+            "previous waveform finished.",
+        )
+
+        self.add_parameter(
+            "trigger_timer_time",
+            get_cmd="TRIGGER:TIMER:TIME?",
+            set_cmd="TRIGGER:TIMER:TIME {:f}",
             get_parser=float,
-            unit='s',
+            unit="s",
             vals=vals.Numbers(100e-9, 20),
-            docstring='Triggering period when in timer trigger mode')
+            docstring="Triggering period when in timer trigger mode",
+        )
 
         self.add_parameter(
-            'trigger_timer_delay',
-            get_cmd='TRIGGER:TIMER:DELAY?',
-            set_cmd='TRIGGER:TIMER:DELAY {:d}',
+            "trigger_timer_delay",
+            get_cmd="TRIGGER:TIMER:DELAY?",
+            set_cmd="TRIGGER:TIMER:DELAY {:d}",
             get_parser=int,
-            unit='1/sample rate',
-            vals=vals.Multiples(min_value=152, max_value=8000000, divisor=8))
+            unit="1/sample rate",
+            vals=vals.Multiples(min_value=152, max_value=8000000, divisor=8),
+        )
```

Comment:
Add docstring for this and `trigger_delay` because it's not obvious what the difference is.

Reply:
done